app/reports/subscription_report.rb

```diff
@@ -0,0 +1,41 @@
+class SubscriptionReport
+  def initialize(subscription, columns)
+    @subscription = subscription
+    @user         = subscription.user
+    @plan         = subscription.plan
+    @columns      = columns
+    @models       = { subscription: subscription, user: user, plan: plan }
+
+    initialize_methods
+  end
+
+  def to_row
+    [].tap do |row|
+      columns.keys.each do |column|
+        columns[column].each { |c| row << send("#{column}_#{c}") }
+      end
+    end
+  end
+
+  private
+
+  attr_reader :subscription, :user, :plan, :columns, :models
+
+  def initialize_methods
+    %i(subscription user plan).each do |model|
+      columns.fetch(model, []).each do |attr|
+        create_method("#{model.to_s}_#{attr.to_s}") { models[model].send(attr) }
```

Comment:
😱 

This is VERY dangerous! Why don't you use simple code instead of the meta-programming?